### ISSUE TITLE
One letter 'd' argument is not working after upgrade to .NET 6

### ISSUE THREAD

**Issue by rafal-mz** [dotnet/sdk#23003]
Our templates have 'description' argument which shorter version is 'd'. 
We used this pattern also for our different arguments, so our templates usually looks like:

`dotnet new our-team-template -f netcore -d "my project" -s "something else"`

It was working like a charm until we migrated to .NET 6. Currently we get an error, when we specify short version of 'd' argument.

![image](https://user-images.githubusercontent.com/45458298/144456912-41c9ab65-31e6-42eb-9c3a-c1487c2f9611.png)

We added stuff in configured params in the following way.

![image](https://user-images.githubusercontent.com/45458298/144457222-476991f7-1250-4dd0-bdb6-84d583fc6147.png)

![image](https://user-images.githubusercontent.com/45458298/144457335-1096eb81-4931-4e50-a206-299f14a98790.png)

Is it something we are missing or this is bug on your side?

**Comment by vlada-shubina** [comment]
Transferring to dotnet/sdk. 
The code below removes all tokens matching diagnostic aliases prior to sending arguments to `dotnet new`:
https://github.com/dotnet/sdk/blob/9d64b48802348f06cf7499ed488ceed0f8572128/src/Cli/dotnet/ParseResultExtensions.cs#L77-L92

A possible way to fix it to remove aliases only if it's given before subcommand.

**Comment by baronfel** [comment]
Yep, we need to change `GetSubArguments` to only strip Diag options between the .NET host (`dotnet`) token and the first command (`new` in this case, but could be anything).

We don't control the `--diag` option, it's coming from the runtime, and they control the actual parsing of it, so we are just responding to that constraint.

**Comment by marcpopMSFT** [comment]
Old issue triage: Appears to work now and I think it always worked if applied right after the dotnet command.
